File: 2ºAÑO/1ºCuatrimestre/PDOO/Civitas/Civitas/src/civitas/Dado.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package civitas;

import java.util.Random;

/**
 *
 * @author Jose
 */
class Dado {
    static final private Dado instance = new Dado();
    
    private Random random;
    private int ultimoResultado;
    private Boolean debug;
    
    static private int SalidaCarcel = 5;
    
    public static Dado getInstance() {
        return instance;
    }
    
    private Dado() {
        random = new Random();
        ultimoResultado = 0;
        debug = false;
    }
    
    int tirar() {
        if(debug == false) {
            ultimoResultado = (random.nextInt(6)+1);
        }
        
        else {
            ultimoResultado = 1;
        }
        
        return ultimoResultado;
    }
    
    Boolean salgoDeLaCarcel() {
        int valor = this.tirar();
        System.out.println("Valor dado = " + valor);
        //esto de cout creo que está mal
        if(valor >= 5) {
            return true;
        }
        
        else {
            return false;
        }
    }
    
    int quienEmpieza(int n) {
        return (random.nextInt(n));
    }
    
    void setDebug(Boolean d) {
        debug = d;
    }
    
    int getUltimoResultado() {
        return ultimoResultado;
    }
}

File: 2ºAÑO/1ºCuatrimestre/PDOO/examen_p1/examen/pkg/Civitas_ruby-0.0.1/lib/dado.rb
# encoding:utf-8

require 'singleton'
module Civitas
  class Dado
    include Singleton
    
    @@SalidaCarcel = 5
    
    def initialize
      @random = Random.new
      @ultimo_resultado = 0
      @debug = false
    end
    #este constructor es privado??
        
    def tirar
      if !@debug
        @ultimo_resultado = @random.rand(5) + 1
      else
        @ultimo_resultado = 1
      end
      return @ultimo_resultado
    end
    
    def salgo_de_la_carcel
      valor = tirar
      puts "ha salido el numero #{valor}"
      if valor >=5
        return true
      else
        return false
      end
    end
    
    def quien_empieza(n)
      return rand(n)
    end
    
    def debug=(d)
      @debug = d
    end
    
    def get_ultimo_resultado
      return @ultimo_resultado
    end  
    
  end
end